### ISSUE TITLE
Using Hardware Acceleration with Docker

### ISSUE THREAD

**Issue by adattudos** [team-inrs/CarND-Capstone#2]
A [Dockerfile](https://github.com/team-inrs/CarND-Capstone/blob/develop/Dockerfile) is provided for the Capstone project to run ROS in a container which is both hardware-agnostic and platform-agnostic. This is obviously not the case when using NVIDIA GPUs since it is using specialized hardware and it requires the installation of the NVIDIA driver. As a result, Docker Engine does not natively support NVIDIA GPUs with containers but [nvidia-docker](https://github.com/NVIDIA/nvidia-docker) provides a solution.

The ROS wiki has a tutorial on [Using Hardware Acceleration with Docker](http://wiki.ros.org/docker/Tutorials/Hardware%20Acceleration).

I think this might be useful in case we want to use GPU accelerated TensorFlow in our project.

What do you think?

**Comment by adattudos** [comment]
I have created a [Dockerfile.gpu](https://github.com/adattudos/CarND-Capstone/blob/develop/Dockerfile.gpu) that seems to work with the project. Shall I submit a pull request?

**Comment by wmrifenb** [comment]
Sure! Nice work. Submit a pull request to Udacity as well. They may like it.

**Comment by adattudos** [comment]
I have played around with nvidia-docker and the original Capstone Dockerfile. Using GPU acceleration for [tensorflow requires](https://www.tensorflow.org/install/install_linux) CUDA Toolkit and cuDNN. Adding this to the Dockerfile would be slightly inconvenient. The other approach is use nvidia/cuda:8.0-cudnn6-devel-ubuntu16.04 as base image and add ROS packages and other requirements.

In fact this is what has been [proposed by @tanthony](https://discussions.udacity.com/t/docker-image-with-cuda-enabled/383576). Their [CarND-Capstone-Docker](https://github.com/SiliconCar/CarND-Capstone-Docker) achieves the goal of providing hardware GPU acceleration for the Capstone project.

I have tested it and it seems to work with basic Tensorflow scripts. Even though the image is not as lightweight as I would want (6 GB) but I propose that we could start working with that.
